pos_discount_absolute/static/src/js/pos_discount_absolute.js

```diff
@@ -0,0 +1,146 @@
+odoo.define('pos_discount_absolute', function (require) {
+"use strict";
+
+    var models = require('point_of_sale.models');
+    var gui = require('point_of_sale.gui');
+    var Widget = require('web.Widget');
+    var PosBaseWidget = require('point_of_sale.BaseWidget');
+    var PopupWidget = require('point_of_sale.popups');
+    var core = require('web.core');
+    var screens = require('point_of_sale.screens');
+
+    var QWeb = core.qweb;
+
+
+    PosBaseWidget.include({
+        init:function(parent,options){
+            var self = this;
+            this._super(parent,options);
+            if (this.gui && this.gui.screen_instances.products &&
+                this.gui.screen_instances.products.action_buttons.discount && this.pos.config.discount_abs_on) {
+                var disc_widget = this.gui.screen_instances.products.action_buttons.discount;
+                disc_widget.button_click = function () {
+                    self.gui.show_popup('number', {
+                        'title': 'Absolute Discount',
+                        'value': self.pos.config.discount_abs_value,
+                        'confirm': function (val) {
+                            self.apply_abs_discount(val);
+                            self.pos.config.discount_abs_value = val;
+                        },
+                    });
+                };
+            }
+        },
+        apply_abs_discount: function(val) {
+            var order = this.pos.get_order();
+            var lines = order.get_orderlines();
+            var product = this.pos.db.get_product_by_id(this.pos.config.discount_product_id[0]);
+            // Remove existing discounts
+            lines.forEach( function(line){
+                if (line.get_product() === product){
+                    order.remove_orderline(line);
+                }
+            });
+            // Add discount
+            if (val != 0){
+                if (this.pos.config.discount_abs_type){
+                    var discount = - Math.min(val, order.get_total_with_tax());
+                    order.add_product(product, { price: discount });
+                } else {
+                    var discount = - val / 100.0 * order.get_total_with_tax();
```

Comment:
This function is called ``apply_abs_discount`` but you use it as relative discount